enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -1017,6 +1025,17 @@ func (c *FirecrackerContainer) LoadSnapshot(ctx context.Context) error {
 	if err != nil {
 		return status.WrapError(err, "failed to get snapshot")
 	}
+
+	// Set unique per-run identifier on the vm metadata so this exact snapshot
+	// run can be identified
+	if snap.GetVMMetadata() == nil {
+		md := &repb.VMMetadata{
+			VmId:       c.id,
+			SnapshotId: c.id,
```

Comment:
Done. In hindsight we should've generated a unique field for VM ID instead. Because c.id is unique per firecracker run, which is more aligned to what snapshotID is than vmID.

Reply:
Yeah, agreed